tools/halide_image.h

```diff
@@ -223,9 +227,14 @@ class Image {
         contents->buf.min[2] = z;
         contents->buf.min[3] = w;
     }
+
+    void info(const char *tag = "Image") const;
+    void dump(const char *tag = "Image") const;
```

Comment:
It's fine if the standalone functions are templated, calls to these functions should be able to deduce the template type no problem...